yesod-core/Yesod/Core/Class/Yesod.hs

```diff
@@ -835,9 +835,20 @@ fileLocationToString loc = (loc_package loc) ++ ':' : (loc_module loc) ++
 --
 -- Since 1.4.16
 guessApproot :: Approot site
-guessApproot = ApprootRequest $ \_master req ->
+guessApproot = guessApprootOr ApprootRelative
+
+-- | Guess the approot based on request headers, with fall back to the
+-- specified 'AppRoot'.
+--
+-- Since 1.4.16
+guessApprootOr :: Approot site -> Approot site
+guessApprootOr fallback = ApprootRequest $ \_master req ->
```

Comment:
Minor point: underscore isn't necessary since the variable is actually being used, e.g. `s/_master/master`